pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/PersistentTopics.java

```diff
@@ -1500,11 +1492,13 @@ private void validateClientVersion() {
                 if (splits != null && splits.length > 1) {
                     if (LEAST_SUPPORTED_CLIENT_VERSION_PREFIX.getMajorVersion() > Integer.parseInt(splits[0])
                             || LEAST_SUPPORTED_CLIENT_VERSION_PREFIX.getMinorVersion() > Integer.parseInt(splits[1])) {
-                        throw new UnsupportedOperationException("version " + userAgent + " is not supported");
+                        throw new RestException(Status.METHOD_NOT_ALLOWED,
+                                "Client lib is not compatible to access partitioned metadata: version " + userAgent
+                                        + " is not supported");
                     }
                 }
-            } catch (UnsupportedOperationException ue) {
-                throw ue;
+            } catch (RestException re) {
+                throw re;
             } catch (Exception e) {
```

Comment:
If validate client version is on and you can't parse the client version - shouldn't we disallow the request? I mean just convert the Exception to a RestException and throw it.

Reply:
here, we are trying to target a specific version which fulfills two conditions: (1) empty user-agent (2) `Pulsar-CPP-v` version < 1.21. So, we know all old clients will fulfill these conditions. but in future with the new-client if parsing fails with new different format then we don't want to reject the request but we want to let it pass. So, I think we should not reject if it can't parse the version because then it must be a new client.